arrow-ord/src/cmp.rs

```diff
@@ -702,4 +772,122 @@ mod tests {
 
         neq(&col.slice(0, col.len() - 1), &col.slice(1, col.len() - 1)).unwrap();
     }
+
+    #[test]
+    fn test_struct_equality() {
+        // test struct('a', 'b') = struct('a', 'b'), the null buffer is 0b0111
+        let left_a = Arc::new(Int32Array::new(
+            vec![0, 1, 2, 3].into(),
+            Some(vec![true, false, true, false].into()),
+        ));
+        let right_a = Arc::new(Int32Array::new(
+            vec![0, 1, 2, 3].into(),
```

Comment:
I think the test coverage would be improved if `right_a` and `left_a` had some different values 

Like maybe

```suggestion
            vec![0, 1, 20, 3].into(),
```

And a similar change below for `right_b` `left_b`